cpp/cmake/modules/ConfigureCUDA.cmake

```diff
@@ -6,7 +6,10 @@
 # =============================================================================
 
 if(CMAKE_COMPILER_IS_GNUCXX)
-  list(APPEND CUML_CXX_FLAGS -Wall -Werror -Wno-unknown-pragmas)
+  # TODO(hcho3): Remove Wno-free-nonheap-object once GCC 15 is released. The flag is needed to work
+  # around spurious warnings emitted by GCC 14.x. See
+  # https://github.com/rapidsai/cuml/pull/7471#issuecomment-3525796585 for more details.
+  list(APPEND CUML_CXX_FLAGS -Wall -Werror -Wno-unknown-pragmas -Wno-free-nonheap-object)
```

Comment:
Disabling `free-nonheap-object` warning for now, to work around the compiler bug in GCC 14.x.